docs/installation/local.md

```diff
@@ -2,169 +2,177 @@
+  + To let your personal fork up-to-date with the original FOSSASIA repo, add original repo to "remote" list and regularly fetch its new content.
 
-**Note:** For Mac OS Sierra users, if you get an error that 'openssl/aes.h' could not be found when installing requirements.txt using pip, try the steps shown here - [OSX openssl header error](https://tutorials.technology/solved_errors/1-OSX-openssl_opensslv_h-file-not-found.html)
+  ```sh
+  git remote add upstream https://github.com/fossasia/open-event-server.git
+  git fetch upstream
+  git merge upstream/development
+  ```
 
-# Installation in Virtual Environment
+## Install system dependencies
 
-You can use either **pip** or **pipenv** to install Open Event Server in a virtual environment.
+These are softwares on which our Open Event server depends, and C-based libraries on which our Python packages depend.
 
-Firstly, open a terminal and enter
+### For Mac OS
 
 ```sh
-# For linux users
-sudo apt-get install python3-dev
-sudo apt-get install libpq-dev
-sudo apt-get install libffi6 libffi-dev
-
-# For macOS users
+brew install postgresql
 brew install python@3
 brew install libmagic
+brew install redis
 ```
 
-## Using pip and virtualenv
+**Note:** For Mac OS Sierra users, if you get an error that 'openssl/aes.h' could not be found when installing requirements.txt using pip, try the steps shown here - [OSX openssl header error](https://tutorials.technology/solved_errors/1-OSX-openssl_opensslv_h-file-not-found.html)
+
+### For Debian/Ubuntu
 
-Open a terminal and enter the following commands to setup a virtual environment
+The dependencies are listed in *deb-packages.txt* file. You can install them all with one command:
 
 ```sh
-sudo apt-get install python3.7
-virtualenv -p python3.7 venv
-. venv/bin/activate
+xargs -a deb-packages.txt sudo apt install
 ```
 
-Now to install the dependencies using pip, type
+In case you use Ubuntu 20.04+, where Python 3.7 is not provided in official repo, you can use [pyenv](https://github.com/pyenv/pyenv) to install Python 3.7 (Open Event Server is not compatible with Python 3.8+ yet).
 
-```sh
-pip3 install -r requirements.txt
-```
+## Create a Python virtual environment
 
-## Using pipenv
+Python virtual environment is a way to keep this project's collection of libraries not interfere with other projects or the system. There are many ways to create Python virtual environment (all have the same core), like:
 
-Using pipenv, you will not need to set up virtualenv. It will do it automatically for you
+- Use Python built-in tool:
 
-To setup a virtual environment and install the dependices, enter in a terminal
+  ```sh
+  python3 -m venv my-project-name
+  ```
 
-```sh
-pipenv --python 3.7.3 install
-```
+- Use third-party tools:
+
+  ```sh
+  mkvirtualenv my-project-name -p /usr/bin/python3
+  ```
+
+I (Quân) recommend [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/) because:
```

Comment:
It's better not to use names in the documentation. @iamareebjamal will be able to tell better, though I am pretty sure these points as well as personal opinions shouldn't be added to README.

Reply:
> if they make a new branch to solve an issue, and if the developer is in the development branch itself, it would just update it

@prabhupant `rebase` means "rewrite Git history" and that causes the mess.
"merge" also let the current branch have update from development branch, but via "merge", not causing "rewrite Git history" and it is safer.

This is from my years of experience, I don't know how to make you understand if you don't stumble upon this circumstance. But as someone who have experience, I need to tell beginner to avoid the bad practice, which just waste time to recover.

To summary:

- "Rebase / Rewrite Git history" is safe if your commits are still in local (not push to remote repo yet).
- "Rebase / Rewrite Git history" will cause mess if you are rewriting the commits which have been pushed before.

Reply:
@prateekj117 I mention my name, just mean that I'm responsible for that guide, so that someone can look for me to ask for more (instead of asking you when you don't recommend that method).

> personal opinions shouldn't be added to README

This is the guide that I want to send to beginners who join this project. If I don't add to README, where should I write? I don't want to let the beginners waste time with not-good way of doing.

Reply:
It's not a not good way of doing it. It's not a bad practice. Almost all tools are compatible with it and none discourage it. Can you show any python doc reference where they discourage to keep virtual environments inside the repo? Even our CI systems like Travis and CircleCI cache the dependencies because they are inside the repo and it is easier to find and make changes in it as well. Keeping it outside the repo is one of the ways of doing and a personal preference. Each way has pro and cons. Keeping it inside can lead to extraneous search results, keeping it outside can make it much more difficult to manage and cache, etx

Reply:
@iamareebjamal 

You keep make mistake between this "local installation guide" and the deployment script. Do the TravisCI, CircleCI tool read this documentation and follow this documentation? Are they human?

> Can you show any python doc reference where they discourage to keep virtual environments inside the repo

Those projects assume that the readers has enough Python knowledge to manage the virtual environment themselves, to resolve issues themselves. They don't tell users to keep virtual environment inside or outside and let it to be personal choice.

The guide I add here is a "tip" for some beginner who may run into problem. Experienced users can skip this, like me, when read a Python project REAME and the author tells using Pipevn, I just skip it.

I mentored some students before, and saw some cases when the students run testcases, the test runner failed because it accidentally scan the virtualenv folders, trying to discovering test cases there, running some code in it and got side effect. So I have to write this, to let other beginners not run into the same problem.

Reply:
We need to keep the installation as simple as possible. We can have your recommendation as an alternative but not as the default command. There is no simple way to manage virtual environments outside the repo. 

> I mentored some students before, and saw some cases when the students ran testcases, the test runner failed because it accidentally scan the virtualenv folders, trying to discovering test cases there, running some code in it and got side effect.

Our testrunner is configured to not fail this. I have also mentored a lot of students for running this project and 80% of the problem boils down to them missing a step in the guide. Some don't start with the correct python version, and some don't initialize a virtual environment. Adding this step as a recommended one will only confuse users and they'll definitely be stuck in problems. We need to keep documentation exact and consistent. I have no problem with adding the recommendation, but I can't approve or merge the PR with it being the default method without it being easy to manage. When people tell me they're having an issue installing things, I ask them to `source .venv/bin/activate`. If they have more issues, I tell them to `rm -rf .venv` and repeated. Your recommendation will make it difficult for me to guide students on how to setup things because I will need to ask "Where is your virtual environment?" and then change my commands according to their environment every time. Many people come from JS background, they already have an idea of a per-project dependency environment like `node_modules`.

> The guide I add here is a "tip" for some beginner who may run into problem. Experienced users can skip this, like me, when reading a Python project README and the author tells using Pipevn, I just skip it.

And this is what I am saying, an expert will probably use your method. A beginner will most definitely be confused about what to do and fail in the next step. Pragmatism and practicality beats perfectionism. When we switch to poetry, we can add the section where the entire virtual environment is managed by poetry. Not now.

Reply:
I agree to make change as you suggest, to let communicate with beginner easier.